core/chains/evm/txmgr/evm_inmemory_store_test.go

```diff
@@ -0,0 +1,70 @@
+package txmgr_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	evmtxmgr "github.com/smartcontractkit/chainlink/v2/core/chains/evm/txmgr"
+)
+
+// assertTxEqual asserts that two transactions are equal
+func assertTxEqual(t *testing.T, exp, act evmtxmgr.Tx) {
+	assert.Equal(t, exp.ID, act.ID)
+	assert.Equal(t, exp.IdempotencyKey, act.IdempotencyKey)
+	assert.Equal(t, exp.Sequence, act.Sequence)
+	assert.Equal(t, exp.FromAddress, act.FromAddress)
+	assert.Equal(t, exp.ToAddress, act.ToAddress)
+	assert.Equal(t, exp.EncodedPayload, act.EncodedPayload)
+	assert.Equal(t, exp.Value, act.Value)
+	assert.Equal(t, exp.FeeLimit, act.FeeLimit)
+	assert.Equal(t, exp.Error, act.Error)
+	assert.Equal(t, exp.BroadcastAt, act.BroadcastAt)
+	assert.Equal(t, exp.InitialBroadcastAt, act.InitialBroadcastAt)
+	assert.Equal(t, exp.CreatedAt, act.CreatedAt)
+	assert.Equal(t, exp.State, act.State)
+	assert.Equal(t, exp.Meta, act.Meta)
+	assert.Equal(t, exp.Subject, act.Subject)
+	assert.Equal(t, exp.ChainID, act.ChainID)
+	assert.Equal(t, exp.PipelineTaskRunID, act.PipelineTaskRunID)
+	assert.Equal(t, exp.MinConfirmations, act.MinConfirmations)
+	assert.Equal(t, exp.TransmitChecker, act.TransmitChecker)
+	assert.Equal(t, exp.SignalCallback, act.SignalCallback)
+	assert.Equal(t, exp.CallbackCompleted, act.CallbackCompleted)
+
+	require.Len(t, exp.TxAttempts, len(act.TxAttempts))
+	for i := 0; i < len(exp.TxAttempts); i++ {
+		assertTxAttemptEqual(t, exp.TxAttempts[i], act.TxAttempts[i])
+	}
+}
+
+func assertTxAttemptEqual(t *testing.T, exp, act evmtxmgr.TxAttempt) {
+	assert.Equal(t, exp.ID, act.ID)
+	assert.Equal(t, exp.TxID, act.TxID)
+	assert.Equal(t, exp.Tx, act.Tx)
+	assert.Equal(t, exp.TxFee, act.TxFee)
+	assert.Equal(t, exp.ChainSpecificFeeLimit, act.ChainSpecificFeeLimit)
+	assert.Equal(t, exp.SignedRawTx, act.SignedRawTx)
+	assert.Equal(t, exp.Hash, act.Hash)
+	assert.Equal(t, exp.CreatedAt, act.CreatedAt)
+	assert.Equal(t, exp.BroadcastBeforeBlockNum, act.BroadcastBeforeBlockNum)
+	assert.Equal(t, exp.State, act.State)
+	assert.Equal(t, exp.TxType, act.TxType)
+
+	require.Equal(t, len(exp.Receipts), len(act.Receipts))
+	for i := 0; i < len(exp.Receipts); i++ {
+		assertChainReceiptEqual(t, exp.Receipts[i], act.Receipts[i])
```

Comment:
Same question as above here but it may not be relevant for this if we're expecting only one receipt per attempt